internal/ir/xds.go

```diff
@@ -176,7 +177,7 @@ type HTTPRoute struct {
 	// Redirections to be returned for this route. Takes precedence over Destinations.
 	Redirect *Redirect
 	// Destinations associated with this matched route.
-	Destinations []*RouteDestination
+	DestinationCluster *DestinationCluster
```

Comment:
would it be easier to guarantee reuse as well as uniqueness if we added a 
`Destination string` here and created a top level `Destinations map[string][]*RouteDestination` within `Xds` ?

Reply:
yeah that's a better approach, I will update the PR.